### DISCUSSION TITLE
multiple requirements.txt files

### DISCUSSION THREAD

**Discussion by jschwar313** [community/community#59972] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have more than one version of my code and they have different folders in GitHub.  In the past, I've been notified of security issues with a version of a python package in my requirements.txt.  Will I still have that ability now that I have more than one requirements.txt?  Or does it have to be located in a certain location to get such notifications?  Thanks.

**Comment by GhostOf0days** [comment, marked as answer]
To get this desired behavior, use [CodeQL Analysis](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/customizing-code-scanning). However, do not store your code like that.Store the main version of your code on one branch, use ```git checkout -b insertNewBranchName```. This creates a new branch and switches the repo you're on to that branch (the '-b' is what creates a new branch). Put another version of your code in this branch and push the changes. ```git checkout insertExistingBranchName``` is what switches branches without creating a new one under that name. After committing another version of your code on the new aforementioned branch, switch to your main branch with ```git checkout main``` then repeat this process for as many versions as you need. Add the branch names to scan as an array like this within the configuration of your codeql-analysis.yml file:     
branches: [main, protected]. Also, the configuration of the file is specific, so please look into [how to set it up](https://medium.com/mycostech/lets-analyze-your-code-repository-in-github-via-codeql-engine-dd196354f77e).

**Reply by GhostOf0days** [reply]
You will have to install [Git](https://git-scm.com/downloads) for this.

**Comment by jschwar313** [comment]
Yea, I thought that's what you'd say.  Thanks.  I do have git installed.

**Reply by GhostOf0days** [reply]
happy to help

**Comment by zxcvb51226** [comment]
> ### Select Topic Area
> 
> Question
> 
> ### Body
> 
> I have more than one version of my code and they have different folders in GitHub.  In the past, I've been notified of security issues with a version of a python package in my requirements.txt.  Will I still have that ability now that I have more than one requirements.txt?  Or does it have to be located in a certain location to get such notifications?  Thanks.

**Comment by jschwar313** [comment]
This can be closed.

**Comment by jschwar313** [comment]
I have one more question.  How do I put the version of python in my requirements.txt file so that GitHub would tell me when there is a security vulnerability with that version of python?

**Reply by GhostOf0days** [reply]
Vulnerabilities are detected for dependencies not Python itself. List dependencyName=version.
